src/front/glsl/variables.rs

```diff
@@ -92,6 +127,8 @@ impl Program {
             Ok(Some(local_var))
         } else if let Some(global_var) = self.context.lookup_global_var_exps.get(name) {
             Ok(Some(*global_var))
+        } else if let Some(global_member) = self.lookup_global_struct_member(name) {
```

Comment:
Would it be possible, instead of calling `lookup_global_struct_member` here, make sure these fields are present in the `lookup_global_var_exps` table. I.e. once we encounter an anonymous global struct, just make sure we add the members to that map, and avoid any wacky lookups later on.

Reply:
Will take a look 👀
First step was to make it work :)

Reply:
Done. Simpler code - though a bit sad to see all my fancy FP code gone 😁

Reply:
Done. Simpler code - a bit sad to see all my fancy FP fun gone though 😁